### ISSUE TITLE
pbskera segmentation fault

### ISSUE THREAD

**Issue by chenxf611** [PacificBiosciences/pbbioconda#475]
**Operating system**
Which operating system and version are you using?

CENTOS_MANTISBT_PROJECT="CentOS-7"

**Package name**
Which package / tool is causing the problem? Which version are you using, use `tool --version`. Have you updated to the latest version `conda update package`? Have you updated the complete env by running `conda update --all`? Have you ensured that your channel priorities are set up according to the bioconda recommendations at https://bioconda.github.io/#set-up-channels?

pbskera

**Conda environment**
What is the result of `conda list`? (Try to paste that between [triple backticks](https://help.github.com/articles/creating-and-highlighting-code-blocks/).)

Name                    Version                   Build  Channel
isoseq3                   3.4.0                         0    bioconda
pbskera                   0.0.100              hdfd78af_0    bioconda

**Describe the bug**

skera -j 8 SW1753.ccs.bam primers.fa SW1753_out.bam
Segmentation fault

I changed the -j number from 36 to 24 to 16 to 8, and increase mem size to 128g, every time it processed 120K to 360K reads and then segmentation fault. 

 

**Error message**
Paste the error message / stack.

Segmentation fault

**To Reproduce**
Steps to reproduce the behavior. Providing a minimal test dataset on which we can reproduce the behavior will generally lead to quicker turnaround time!

I tried with two different ccs.bam files and got same error.

**Expected behavior**

Thanks 

Jack

**Comment by zeeev** [comment]
Hi @chenxf611,

Thanks for the bug report! Please try adding the algorithm option. We've already changed the default, but haven't released the fix yet. 

```
skera --algorithm kmer <movie>.hifi_reads.bam adapters.fasta <movie>.skera.bam
````

**Comment by chenxf611** [comment]
Thanks Zev, it generated the output.bam and it's fast!

I have another question, the seq id seems to be the same for the reads from the same ccs read, according to the description in "Segmented reads",  the read name should be "{movieName}/{holeNumber}/ccs/{qStart}_{qEnd}", however, from my output.bam, the name "m64235e_220214_203014/1/ccs" appeared for 13 reads, without the "{qStart}_{qEnd}" part, I don't know if I missed something.

Thanks

Jack

**Comment by zeeev** [comment]
Hi @chenxf611  (Jack),

We've updated to version [102](https://github.com/bioconda/bioconda-recipes/pull/33448), which it not yet merged. It will be even faster than before. 

I think I saw you mention read names. The code is still in alpha release meaning we haven't finished all the features. The renames will be updated to match the documents, soon.

Best wishes, and thank you for trying out our latest tool,

Zev
